Connectors/Rhino/Speckle.Connectors.RhinoShared/Bindings/RhinoMapperBinding.cs

```diff
@@ -171,18 +326,37 @@ private void OnObjectChanged(object? sender, RhinoObjectEventArgs e)
   /// <summary>
   /// Called when object attributes are modified in Rhino.
   /// </summary>
+  /// <remarks>
+  /// Includes detection for when objects move between layers with mappings.
+  /// </remarks>
   private void OnObjectAttributesChanged(object? sender, RhinoModifyObjectAttributesEventArgs e)
   {
+    if (!_store.IsDocumentInit)
+    {
+      return;
+    }
+
     var rhinoObject = e.RhinoObject;
-    if (!string.IsNullOrEmpty(rhinoObject.Attributes.GetUserString(CATEGORY_USER_STRING_KEY)))
+
+    // check if mapping user string changed (added, removed, or modified)
+    var oldMapping = e.OldAttributes.GetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY);
+    var newMapping = rhinoObject.Attributes.GetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY);
+    bool mappingChanged = !string.Equals(oldMapping, newMapping, StringComparison.Ordinal);
+
+    // check if layer change affects mappings
+    bool hasOldLayerMapping = HasLayerMapping(e.OldAttributes.LayerIndex);
+    bool hasNewLayerMapping = HasLayerMapping(rhinoObject.Attributes.LayerIndex);
+
+    // refresh if mapping changed OR layer change affects mapped layers
+    if (mappingChanged || hasOldLayerMapping || hasNewLayerMapping)
     {
       _idleManager.SubscribeToIdle(nameof(NotifyMappingsChanged), NotifyMappingsChanged);
     }
   }
```

Comment:
we know already when mappings has changed on UI, why we track this over rhino events api?